test/units/cli/test_galaxy.py

```diff
@@ -115,6 +117,32 @@ def test_display_galaxy_info(self):
         if display_result.find('\t\tgalaxy_tags:') > -1:
             self.fail('Expected galaxy_tags to be indented twice')
 
+    def test_execute_install(self):
+        # testing installing with insufficient information
+        gc = GalaxyCLI(args=["install"])
+        with patch('sys.argv', ["-c"]):
```

Comment:
I think this indicates GalaxyCLI is doing something odd with it's args. I would expect passing args to GalaxyCLI to pass the value of args to GalaxyCLI's self.parser.parse_args instead of of calling it as self.parser.parse_args() which defaults to looking at sys.argv.

ie, change GalaxyCLI.parse to take args
``` python
def parse(self, args=None):
```
and then have GalaxyCLI.parse call:
``` python
self.options, self.args =self.parser.parse_args(args)
```

Then the test would just pass ['install', '-c'] as args instead of having to patch sys.argv so the testing
is easier and less likely to cause odd side effects, especially if anything else is looking at sys.argv (like nosetest
itself for example).

If the top level app also passes in a copy of sys.argv instead of a reference (ie, GalaxyCLI(args=sys.argv[:])), it avoids problems potentially caused by app code manipulating sys.argv.

That said, it's been doing that a long time, so doesn't need to be changed for this pr, but it is an example of a case where changing the code to be more testable is useful.